doc/manual/source/release-notes/rl-2.28.md

```diff
@@ -0,0 +1,109 @@
+# Release 2.28.0 (2025-04-02)
+
+This is an atypical release -- instead of being branched off from `master`, it is branched off from the 2.27.x maintenance branch.
+The purpose of this is to satisfy both these goals:
+
+- Release with number of API-breaking changes that are not suitable to backport to 2.27
+
+- Do not Release with arbitrary new commits from master
+
+The reason for the combinations of these goals is that we would like this version of Nix to the default in Nixpkgs 25.05, yet, we are getting close to the Nixpkgs 25.05 version freeze.
+These API changes complete the big infrastructure rework that accompanies the switch to Meson --- we want to batch all these changes together so there is one round of breakage.
+But we don't want to to release with arbitrary new changes form master, so close to a major release, before those changes have had time to "incubate".
```

Comment:
```suggestion
This is an atypical release, and for almost all intents and purposes, it is just a continuation of 2.27; not a feature release.

We had originally set the goal of making 2.27 the Nixpkgs default for NixOS 25.05, but dependents that link to Nix need certain _interface breaking_ changes in the C++ headers. This is not something we should do in a patch release, so this is why we branched 2.28 right off 2.27 instead of `master`.

This completes the infrastructure overhaul for the [RFC 132](https://github.com/NixOS/rfcs/blob/master/rfcs/0132-meson-builds-nix.md) switchover to meson as our build system.
```